pkg/apis/rollouts/v1alpha1/analysis_types.go

```diff
@@ -61,6 +63,21 @@ type Metric struct {
 	Provider AnalysisProvider `json:"provider"`
 }
 
+// EffectiveCount is the effective count based on whether or not count/interval is specified
+// If neither count or interval is specified, the effective count is 1
+// If only interval is specified, metric runs indefinitely and there is no effective count (nil)
+// Otherwise, it is the user specified value
+func (m *Metric) EffectiveCount() *int32 {
+	if m.Count == 0 {
+		if m.Interval == nil {
+			one := int32(1)
+			return &one
```

Comment:
minor: you can use pointer.Int32Ptr(1) from the `	"k8s.io/utils/pointer"` library

Reply:
I considered it, but wanted to reduce the import burden for anyone importing our types.